### PR TITLE
kinesis support

### PR DESCRIPTION
**Pull request by ConradKurth**
We would like to push directly to kinesis and thought others would like that support

### CODE REVIEW

**Comment by ConradKurth** [comment]
@zxan1285 can i get some comments on this?

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### auth0-logs-to-splunk v2.2.0
 ### auth0-logs-to-sumologic v2.1.0
 ### logs-to-cloudwatch v2.1.0
+## logs-to-kinesis v1.0.0
```
Comment (zxan1285) [inline CHANGELOG.md]: Please, remove that. We're not sure about release date yet. We will update changelog on release.

File: server/lib/senders/kinesis.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+
+  return (logs, callback) => {
+    if (!logs || !logs.length) {
+      return callback();
+    }
+
+    logger.info(`Sending ${logs.length} logs to Kinesis...`);
+    // these are the max number of records that can be sent to kinesis
+    const chunks = chunk(logs, maxRecords);
```
Comment (zxan1285) [inline server/lib/senders/kinesis.js]: This seems unnecessary. Max BATCH_SIZE is 100, so we cannot get more than 100 logs here.

File: CHANGELOG.md (hunk 1 above)
Comment (ConradKurth) [reply CHANGELOG.md]: Done

File: server/lib/senders/kinesis.js (hunk 2 above)
Comment (ConradKurth) [reply server/lib/senders/kinesis.js]: Is that the `max` size or is that just the default that shows up in the UI? It seems like you can make that larger in the UI to something more than 500. Kinesis has a limit of sending more than 500 records at a time, https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Kinesis.html#putRecords-property

Comment (zxan1285) [reply server/lib/senders/kinesis.js]: https://github.com/auth0-extensions/auth0-logs-to-provider/blob/master/server/lib/processLogs.js#L74
100 is the maximum for all extensions but mixpanel.

Comment (ConradKurth) [reply server/lib/senders/kinesis.js]: @zxan1285 Gotcha, made the changes!

File: server/lib/senders/kinesis.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  const kinesis = new AWS.Kinesis({ apiVersion: '2013-12-02' });
+  return (logs, callback) => {
+    if (!logs || !logs.length) {
+      return callback();
+    }
+
+    logger.info(`Sending ${logs.length} logs to Kinesis...`);
+
+    logs.forEach(log => {
+      const records = log.map((log) => {
```
Comment (zxan1285) [inline server/lib/senders/kinesis.js]: `log` is an object here, not an array.
I assume, we don't need `logs.forEach` here, just `logs.map(log => ...)`.

Comment (ConradKurth) [reply server/lib/senders/kinesis.js]: @zxan1285 Ahh right, was moving too quick. done

**Comment by ConradKurth** [comment]
@zxan1285 is there a time frame on when this would be integrated into auth0?

**Comment by zxan1285** [comment]
@ConradKurth unfortunately, i cannot provide any ETA on this.

At this point i can only recommend you to install this extension as custom. For doing so, you need to build it (`npm run build`), add bundle file (`dist/kinesis/logs-to-kinesis.extension.1.0.0.js`) and corresponding `webtask.json` to the repository, add `codeUrl` property to that `webtask.json` file like this - https://github.com/auth0-extensions/auth0-deploy-cli-extension/blob/master/webtask.json#L12. After that you can hit `create extension` button in the extensions dashboard and paste url of `webtask.json` file of your new extension.

**Pull request closed without merging**